src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetInstaller.cs

```diff
@@ -238,17 +293,23 @@ public Task<UninstallResult> UninstallAsync(IManagedTemplatePackage templatePack
             _ = templatePackage ?? throw new ArgumentNullException(nameof(templatePackage));
             if (!(templatePackage is NuGetManagedTemplatePackage))
             {
-                return Task.FromResult(UninstallResult.CreateFailure(templatePackage, InstallerErrorCode.UnsupportedRequest, $"{templatePackage.Identifier} is not supported by {Factory.Name}"));
+                return Task.FromResult(UninstallResult.CreateFailure(
+                    templatePackage,
+                    InstallerErrorCode.UnsupportedRequest,
+                    string.Format(LocalizableStrings.NuGetInstaller_InstallResut_Error_PackageNotSupported, templatePackage.DisplayName, Factory.Name)));
             }
             try
             {
                 _environmentSettings.Host.FileSystem.FileDelete(templatePackage.MountPointUri);
                 return Task.FromResult(UninstallResult.CreateSuccess(templatePackage));
             }
-            catch (Exception ex)
+            catch (Exception e)
             {
-                _logger.LogDebug($"Uninstalling {templatePackage.DisplayName} failed. Details:{ex.ToString()}");
-                return Task.FromResult(UninstallResult.CreateFailure(templatePackage, InstallerErrorCode.GenericError, $"Failed to uninstall {templatePackage.DisplayName}, reason: {ex.Message}"));
+                _logger.LogDebug($"Uninstalling {templatePackage.DisplayName} failed. Details:{e.ToString()}");
```

Comment:
Not your change, but wanted to add. I think it is better performance-wise to leave the formatting to the `ILogger` like below. If "Debug" level is too low and the message won't be displayed, then the final string won't be constructed.
```suggestion
                _logger.LogDebug($"Uninstalling {0} failed. Details: {1}", templatePackage.DisplayName, e.ToString());
```